app/main/auto-updater/auto-updater.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Auto Updater
+ * 
+ * Similar to https://electronjs.org/docs/api/auto-updater#events
+ * See https://electronjs.org/docs/tutorial/updates for documentation
+ */
+import { app, autoUpdater, dialog } from 'electron';
+import moment from 'moment';
+import B from 'bluebird';
+import { getUpdate } from './update-checker';
+import { getFeedUrl } from './config';
+  
+autoUpdater.setFeedURL(getFeedUrl(app.getVersion()));
+
+/** 
+ * Check for new updates
+ */
+export async function checkNewUpdates (fromMenu) {
+  // autoupdate.checkForUpdates always downloads updates immediately 
+  // This method (getUpdate) let's us take a peek to see if there is an update 
+  // available before calling .checkForUpdates
+  const update = await getUpdate(app.getVersion());
+  if (update) {
+    let {name, notes, pub_date} = update;
+    pub_date = moment(pub_date).format('MMM Do YYYY, h:mma');
+
+    // Ask user if they wish to install now or later
+    dialog.showMessageBox({
+      type: 'info',
+      buttons: ['Install Now', 'Install Later'],
+      message: `Appium Desktop ${name} is available`,
+      detail: `Release Date: ${pub_date}\n\nRelease Notes: ${notes.replace("*", "\n*")}`,
+    }, (response) => {
+      if (response === 0) {
+        // If they say yes, get the updates now
+        autoUpdater.checkForUpdates();
+      }
+    });
+  } else {
+    if (fromMenu) {
+      autoUpdater.emit('update-not-available');
+    } else {
+      // If no updates found check for updates every hour
+      await B.delay(60 * 1000);
```

Comment:
is the interval in seconds or millis?

Reply:
why don't we use setTimeout?

Reply:
ms; setTimeout does not return a promise and can't be awaited

Reply:
I changed it to `60 * 60 * 1000` so that it checks for updates every hour and not every minute.